StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1008OpeningParenthesisMustBeSpacedCorrectly.cs

```diff
@@ -25,17 +27,49 @@ public class SA1008OpeningParenthesisMustBeSpacedCorrectly : DiagnosticAnalyzer
         /// The ID for diagnostics produced by the <see cref="SA1008OpeningParenthesisMustBeSpacedCorrectly"/> analyzer.
         /// </summary>
         public const string DiagnosticId = "SA1008";
+
+        internal const string LocationKey = "location";
+        internal const string ActionKey = "action";
+        internal const string LocationPreceding = "preceding";
+        internal const string LocationFollowing = "following";
+        internal const string ActionInsert = "insert";
+        internal const string ActionRemove = "remove";
+
         private const string Title = "Opening parenthesis must be spaced correctly";
-        private const string MessageFormat = "Opening parenthesis must{0} be {1} by a space.";
         private const string Category = "StyleCop.CSharp.SpacingRules";
         private const string Description = "An opening parenthesis within a C# statement is not spaced correctly.";
         private const string HelpLink = "http://www.stylecop.com/docs/SA1008.html";
 
-        private static readonly DiagnosticDescriptor Descriptor =
-            new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, Category, DiagnosticSeverity.Warning, AnalyzerConstants.DisabledNoTests, Description, HelpLink);
+        private const string MessageNotPreceded = "Opening parenthesis must not be preceded by a space.";
+        private const string MessagePreceded = "Opening parenthesis must be preceded by a space.";
+        private const string MessageNotFollowed = "Opening parenthesis must not be followed by a space.";
 
         private static readonly ImmutableArray<DiagnosticDescriptor> SupportedDiagnosticsValue =
-            ImmutableArray.Create(Descriptor);
+            ImmutableArray.Create<DiagnosticDescriptor>()
+                .Add(DescriptorNotPreceded)
+                .Add(DescriptorPreceded)
+                .Add(DescriptorNotFollowed);
+
+        /// <summary>
+        /// Gets the diagnostic descriptor for an opening parenthesis that must not be preceded by whitespace.
+        /// </summary>
+        /// <value>The diagnostic descriptor for an opening parenthesis that must not be preceded by whitespace.</value>
+        public static DiagnosticDescriptor DescriptorNotPreceded =>
```

Comment:
:bulb: Use `{ get; } =` instead of `=>`. The way it's written here will cause a new `DiagnosticDescriptor` instance to be created every time this property is accessed.